frontend/src/views/WheresMyGene/components/GeneSearchBar/index.tsx

```diff
@@ -160,35 +211,124 @@ export default function GeneSearchBar({ onGenesChange }: Props): JSX.Element {
+      const newSelectedGenes = [...selectedGenes];
+      const pastedGenes = pull(uniq(input.split(/[ ,]+/)), "");
+      const allGenes = genes.reduce((acc, gene) => {
+        return acc.set(gene.name, gene);
+      }, new Map<Gene["name"], Gene>());
+      pastedGenes.map((gene) => {
+        const newGene = allGenes.get(gene);
+        if (!newGene) {
+          Toast.show({
+            intent: Intent.DANGER,
+            message: `Gene not found: ${gene}`,
+          });
+        } else if (!newSelectedGenes.includes(newGene))
+          newSelectedGenes.push(newGene);
+      });
+      setPendingPaste(false);
+      setOpen(false);
+      return setSelectedGenes(newSelectedGenes);
+    }
+  };
+
+  const handleInput = (
+    _,
+    value: string,
+    reason: AutocompleteInputChangeReason
+  ) => {
+    if (reason === "input") setInput(value);
+  };
+
+  const useStyles = makeStyles((theme: Theme) => {
+    const colors = getColors({ theme });
+    const shadows = getShadows({ theme });
+    const corners = getCorners({ theme });
+    return {
+      paper: {
+        boxShadow: "none",
+        margin: 0,
+      },
+      popper: {
+        backgroundColor: "white",
+        border: `1px solid ${colors?.gray[100]}`,
+        borderRadius: corners?.m,
+        boxShadow: shadows?.m,
+        color: "#586069",
+        fontSize: 13,
+        width: 377,
+        zIndex: 3, // The x axis wrapper is set at 2
+      },
+      popperDisablePortal: {
+        position: "relative",
+        width: "100% !important",
+      },
+    };
+  });
+
+  const classes = useStyles();
+
+  const ref = useRef(null);
 
   return (
     <Container>
       <GeneSets onSelect={handleGenesetsSelect} />
 
       <br />
       <br />
-
-      <label htmlFor="first-n-genes">Select first N genes</label>
-      <br />
-      <input id="first-n-genes" onChange={handleInputChange} value={input} />
-      <MultiSelectWrapper>
-        <MultiSelect
-          itemPredicate={itemPredicate}
-          onItemSelect={handleItemSelect}
-          onRemove={handleItemRemove}
-          items={genes}
-          itemRenderer={renderItem}
-          tagRenderer={TagRenderer}
-          itemsEqual={areGenesEqual}
-          selectedItems={selectedGenes}
-          itemListRenderer={itemListRenderer}
+      <ButtonBase disableRipple onClick={handleClick} ref={ref}>
+        <span>Add Genes</span>
+      </ButtonBase>
+
+      <Popper open={open} className={classes.popper} anchorEl={ref.current}>
+        <MenuSelect
+          open
+          search
+          onClose={handleClose}
+          multiple
+          classes={{
+            paper: classes.paper,
+            popperDisablePortal: classes.popperDisablePortal,
+          }}
+          value={selectedGenes}
+          onChange={handleChange}
+          onInputChange={handleInput}
+          // (seve): chanzuckerberg/sci-components#121
+          placeholder="Search or paste comma separated gene names"
```

Comment:
noop, until issue is addressed

Reply:
Thanks, @seve ! PR out 🎉  https://github.com/chanzuckerberg/sci-components/pull/122